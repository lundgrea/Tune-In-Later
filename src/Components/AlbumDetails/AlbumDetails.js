import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types'

export const AlbumDetails = ({album}) => {
  console.log(album)
  return(
    <article className='details-container'>
      <div className='details'>
        <h1>{album.artist_name}</h1>
        <img src={album.artwork_url} alt=''></img>
        <p> Album: {album.album_name}</p>
        <p>Release Date: {album.release_date}</p>
        <p>Genre: {album.primary_genre_name}</p>
      </div>
      <Link className='detail-button' to='/search'>BACK</Link>
      
    </article>
  )
}

AlbumDetails.propTypes = {
  album: PropTypes.object.isRequired
}
