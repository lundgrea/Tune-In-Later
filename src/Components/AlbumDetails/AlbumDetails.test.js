import React from 'react';
import { shallow } from 'enzyme';
import { AlbumDetails } from './AlbumDetails'

describe('Album Details', () => {
  let wrapper;

  const mockAlbum = {
      artist_name:  'EarthGang',
      album_name:  'MirrorLand',
      primary_genre_name:  'Rap',
      album_id:  12345678,
      artwork_url:  'asdfjaskl;df',
      release_date: '9/05/2019',
      content_advisory_rating: 'Explicit',
      key: 12345678,
      isFavorite: false
  }
  beforeEach(() => {
    wrapper = shallow(
      <AlbumDetails
        album={mockAlbum}
      />
    )
  });
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot() 
  }) 
})