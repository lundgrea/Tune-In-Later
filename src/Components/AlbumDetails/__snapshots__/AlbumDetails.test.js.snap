// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album Details should match snapshot 1`] = `
<article
  className="details-container"
>
  <div
    className="details"
  >
    <h1>
      EarthGang
    </h1>
    <img
      alt=""
      src="asdfjaskl;df"
    />
    <p>
       Album: 
      MirrorLand
    </p>
    <p>
      Release Date: 
      9/05/2019
    </p>
    <p>
      Genre: 
      Rap
    </p>
  </div>
  <Link
    className="detail-button"
    to="/search"
  >
    BACK
  </Link>
</article>
`;
